webhookv3/webhookv3.go

```diff
@@ -105,3 +106,24 @@ func calculateSignature(payload []byte, secret string) []byte {
 	mac.Write(payload)
 	return mac.Sum(nil)
 }
+
+func ReadWebhookPayload(r *http.Request) (*WebhookPayload, error) {
+	orb := r.Body
+
+	b, err := ioutil.ReadAll(io.LimitReader(r.Body, webhookBodyReaderLimit))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	defer func() { _ = orb.Close() }()
+	r.Body = ioutil.NopCloser(bytes.NewReader(b))
+
+	if len(b) == 0 {
+		return nil, ErrMalformedBody
+	}
```

Comment:
> Per the io.Reader interface contract, you should try processing the data in b before you handle the io.ReadAll error.
@theckman : Where are you seeing that? I don't see it mentioned in [io.ReadAll](https://pkg.go.dev/io#ReadAll) or in the example provided there...

(IIRC, we got this code from the comments you provided in the previous PR.)

Reply:
Those snippets were things I probably hammered out on my mobile phone while responding back to the comment(s). In hindsight, I should have done a better job at communicating that I hacked them together pretty rapidly as an example. I'm sorry about that.

Reply:
No worries but I would still like to get clarity on what is the "perfect" reusable HTTP body reader.

This is one of the reasons I was pushing for more focused functions that verify signatures and construct events in https://github.com/PagerDuty/go-pagerduty/pull/326. I would like to focus my effort on providing solid support for V3 Webhooks rather than the details of how folks implement that functionality.